src/Handlers/Bookings/BookingsQueryHandler.php

```diff
@@ -0,0 +1,81 @@
+<?php
+
+namespace RebelCode\EddBookings\RestApi\Handlers\Bookings;
+
+use Dhii\Invocation\InvocableInterface;
+use Exception;
+use Psr\Container\NotFoundExceptionInterface;
+use RebelCode\EddBookings\RestApi\Controller\ControllerInterface;
+use WP_Error;
+use WP_REST_Request;
+use WP_REST_Response;
+
+/**
+ * Handles the REST API endpoint for querying bookings.
+ *
+ * @since [*next-version*]
+ */
+class BookingsQueryHandler implements InvocableInterface
+{
+    /**
+     * The resource controller.
+     *
+     * @since [*next-version*]
+     *
+     * @var ControllerInterface
+     */
+    protected $controller;
+
+    /**
+     * Constructor.
+     *
+     * @since [*next-version*]
+     *
+     * @param ControllerInterface $controller The booking resource controller.
+     */
+    public function __construct(ControllerInterface $controller)
+    {
+        $this->controller = $controller;
+    }
+
+    /**
+     * {@inheritdoc}
+     *
+     * @since [*next-version*]
+     */
+    public function __invoke()
+    {
+        /* @var $request WP_REST_Request */
+        $request = func_get_arg(0);
```

Comment:
I think you should put this into the `try...catch` block, and throw if the arg is not present.